src/task.c

```diff
@@ -26,6 +26,12 @@ static void
 task_system_io_register_thread(task_system *ts);
 // end forward declarations
 
+const char *task_type_name[] = {"TASK_TYPE_INVALID",
+                                "TASK_TYPE_MEMTABLE",
+                                "TASK_TYPE_NORMAL"};
+_Static_assert((ARRAY_SIZE(task_type_name) == NUM_TASK_TYPES),
+               "Array task_type_name[] is incorrectly sized.");
+
```

Comment:
What's the point of the invalid task type? It interacts weirdly with the array in `splinterdb.h` and I don't see the purpose.

Reply:
I did not try to change or correct this. The array sized `[3]` is what's existing in code. We have talked about streamlining this interface to just provide total-number-of-bg-threads, at which time ... these kinds of warts will be automatically resolved.

Hopefully.

Reply:
Nothing to change here ... leave as-is.